include/boost/url/impl/url_base.ipp

```diff
@@ -1337,6 +1337,108 @@ end() const noexcept ->
     return iterator(v_, true);
 }
 
+url_base::segments_type::iterator
+url_base::
+segments_type::
+erase( url_base::segments_type::iterator at ) noexcept
+{
+    auto e = at;
+    return erase(at, ++e);
+}
+
+url_base::segments_type::iterator
+url_base::
+segments_type::
+erase( url_base::segments_type::iterator b, url_base::segments_type::iterator e ) noexcept
+{
+    BOOST_ASSERT(v_ != nullptr);
+    url_base & v = *v_;
+    BOOST_ASSERT(b.v_ == &v);
+    BOOST_ASSERT(e.v_ == &v);
+    BOOST_ASSERT(b.off_ >= v.pt_.offset[detail::id_path]);
+    BOOST_ASSERT(e.off_ >= v.pt_.offset[detail::id_path]);
+    BOOST_ASSERT(b.off_ <= v.pt_.offset[detail::id_query]);
+    BOOST_ASSERT(e.off_ <= v.pt_.offset[detail::id_query]);
+    if( b.off_ == e.off_ ) return b;
+    auto const d = e.off_ - b.off_;
+    BOOST_ASSERT(d > 0);
+    auto slcount = std::count(v.s_ + b.off_, v.s_ + e.off_, '/');
+    BOOST_ASSERT(slcount > 0);
+    BOOST_ASSERT(v.pt_.nseg >= slcount);
+    v.pt_.nseg -= slcount;
+    std::copy(v.s_ + e.off_, v.s_ + v.pt_.offset[detail::id_end] + 1, v.s_ + b.off_);
+    v.pt_.resize(detail::id_path, v.pt_.length(detail::id_path, detail::id_query) - d);
+    BOOST_ASSERT(v.s_[v.pt_.offset[detail::id_end]] == '\0');
+    b.parse();
+    return b;
+}
+
+url_base::segments_type::iterator
+url_base::
+segments_type::
+insert_encoded_segment( segments_type::iterator at, string_view const enseg )
```

Comment:
Arguments passed by value never need to be marked `const`

Reply:
They don't, and should never be, in declarations. In definitions, `const` indicates that the value will not change, which is useful information (I only put it in the definition when appropriate, never in the declaration).

Reply:
I dont use it